tools/publish-aos.js

```diff
@@ -1,54 +1,137 @@
 const Arweave = require('arweave');
 const path = require('path');
 const fs = require('fs');
+const { TurboFactory } = require('@ardrive/turbo-sdk');
+const { createData, ArweaveSigner } = require('@dha-team/arbundles');
 
 const srcCodeTxIdPlaceholder = '__INSERT_SOURCE_CODE_ID__';
 
 const arweave = Arweave.init({
   host: 'arweave.net',
   port: 443,
   protocol: 'https',
+  logging: false,
 });
+
+function getPublishedSourceCodeIds() {
+  const changelog = fs.readFileSync(
+    path.join(__dirname, '../CHANGELOG.md'),
+    'utf-8',
+  );
+
+  const sourceCodeTxIdRegex = /[a-zA-Z0-9-_]{43}/g;
+  const matches = changelog.match(sourceCodeTxIdRegex);
+
+  const sourceCodeTxIds = new Set(matches ?? []);
+
+  return [...sourceCodeTxIds];
+}
```

Comment:
so if i understand this correctly, you're parsing the changelogs to get tx ids, which you then use to validate the latest one has been published? that makes the release script dependent on the changelog, which we'll likely manually modify and manage. also - is the benefit just to avoid publishing the same source code twice? if so, what's the "risk" of doing so if we're gating how and when this script runs, assuming it's part of the CD github actions and not run manually. 

so - do we need to check the previous source code before we merge? and if we assume we'll use gated releases with incremental versioning - won't that process avoid unnecessary deploys?

Reply:
we do not need to do this, removed